Created Workers

Icebearbear opened the pull request:
- installed Pipenv for dependencies management
- installed requests library for HTTP request with pivenv
- get bearer token for Twittter api 
- data fetching from Twitter api in JSON format

Icebearbear committed:
Pipenv installation and Twitter API data fetch
installed Pipenv for dependencies management
installed libraries for HTTP request with pivenv
installation details can be found in README.md

Data fetching is done using requests library in worker.py
before start fetching data, get all the keys and tokens
required for this transaction
all the detailscan be found in README.md

Jaimeloeuf commented on workers/README.md:
``````diff
@@ -0,0 +1,35 @@
+Bearer Token is required to use the Request lib for Twitter API
+`````
+
+`````
``````
Why are there so many empty code blocks in your markdowns? Please change them across the whole repo.
Refer to :
- https://help.github.com/en/github/writing-on-github/basic-writing-and-formatting-syntax
- https://guides.github.com/features/mastering-markdown/

Icebearbear committed:
Created a config file for environment variables
- installed a library called python-detonv to manage the environment
variables files
- edited README.md file

Jaimeloeuf commented on workers/connect_to_postgres.py:
```diff
@@ -0,0 +1,14 @@
+
+def connection() :
+    try :
+        connection_details = "dbname='makeupdb' user='icebear' host='localhost' port='5432'" 
```
Please put the db connection detail values into your env file too before loading it in

Jaimeloeuf commented:
1) Please do not commit your .env file!
    - Purge your env file
    - Add note in README to explain what are the variables needed in the env file, and show a sample .env file in a sample/ directory instead of commiting your private .env file
2) Add a gitignore in repo root too!

Icebearbear committed:
Merged worker.py into connect_to_postgres.py:
1.  deleted worker and moved logic into connect_to_postgres
2.  updated env with postgres connection details
3.  updated README to describe .env
4.  added a section to describe postgres connection and queries into
    README

Icebearbear committed:
Merge pull request #4 from Enkel-Digital/master

merge master to feat/worker for gitignore

Icebearbear committed:
removed .env file from workers/

Jaimeloeuf commented:
Please update code to use logging module instead of prints  
List of references:
- <https://docs.python.org/3/library/logging.html>
- <https://inventwithpython.com/blog/2012/04/06/stop-using-print-for-debugging-a-5-minute-quickstart-guide-to-pythons-logging-module/>
- <https://stackoverflow.com/questions/6579496/using-print-statements-only-to-debug>

Icebearbear committed:
renamed connect_to_postgres to DatabaseConnection

Icebearbear committed:
Used ORM for database connection
1.  installed SQLAlchemy as ORM with pipenv
2.  made a connection to postgres database
3.  inserted the data into database
4.  updated README file to describe the ORM connection

Icebearbear committed:
Environment variables information in README file
- steps to creat and use env file
- dscription of every variables in env file

Jaimeloeuf commented on workers/ORMConnection.py:
```diff
@@ -50,14 +51,14 @@ def data_get_and_insert(search_resp, metadata, tweets, connection) :
 
     else :
-        print('Result for' , search_url , 'is unsuccesful')
+        print('Request unsuccesful')
```
Dont commit changes in ur py file when the commit is about ENV vars in README
